tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("teamdctl {} state item set runner.enable_retry_count_feature false".format(port_channel))
+
+    # Wait 90 seconds for retry count to get updated on the peers
+    time.sleep(90)
+
+    yield
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("teamdctl {} state item set runner.enable_retry_count_feature true".format(port_channel))
+
+
+class TestNeighborRetryCount:
+    def test_peer_retry_count(self, duthost, nbrhosts, higher_retry_count_on_peers):
+        """
+        Test that DUT sees new retry count when peers update retry count.
+        """
+        for nbr in list(nbrhosts.keys()):
+            port_channel_status = nbrhosts[nbr]['host'].get_port_channel_status("PortChannel1")
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5, "retry count on neighbor is incorrect")
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5,
+                          "retry count on DUT is incorrect; expected 5, but is {}"
+                          .format(port_channel_status["runner"]["retry_count"]))
+
+        cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+        port_channels = cfg_facts["PORTCHANNEL"].keys()
+        for port_channel in port_channels:
+            port_channel_status = duthost.get_port_channel_status(port_channel)
+            number_of_lag_member = len(cfg_facts["PORTCHANNEL_MEMBER"][port_channel])
+            pytest_assert("ports" in port_channel_status and number_of_lag_member == len(port_channel_status["ports"]),
+                          "get port status error")
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(status["runner"]["selected"], "lag member is not up")
+                pytest_assert(status["runner"]["partner_retry_count"] == 5,
+                              "partner retry count is incorrect; expected 5, but is {}"
+                              .format(status["runner"]["partner_retry_count"]))
+
+    def test_peer_retry_count_packet_version(self, duthost, nbrhosts, higher_retry_count_on_peers, scapy_lacp_layer):
+        """
+        Test that peers and DUT use new LACPDU version when peers use a non-standard retry count
+        """
+        check_lacpdu_packet_version(duthost)
+
+    def test_kill_teamd_lag_up(self, duthost, nbrhosts, higher_retry_count_on_peers, config_reload_on_cleanup):
+        """
+        Test that the lag remains up for 150 seconds after killing teamd on the peer
+        """
+        for nbr in list(nbrhosts.keys()):
+            nbrhosts[nbr]['host'].shell("sudo pkill -USR1 -x teamd")
+
+        # Give ourselves 15 seconds to check before the LAG goes down.
```

Comment:
Typo here? 15 vs 135

Reply:
This number is 150 seconds-seconds of wait time (150-135=15). This is how long we'll have to verify the LAG state after the sleep.